Sources/NIO/Thread.swift

```diff
@@ -249,56 +144,23 @@ public final class ThreadSpecificVariable<Value: AnyObject> {
     /* the actual type in there is `Box<(ThreadSpecificVariable<T>, T)>` but we can't use that as C functions can't capture (even types) */
     private typealias BoxedType = Box<(AnyObject, AnyObject)>
 
-    private class Key {
-#if os(Windows)
-        private var value: DWORD
-#else
-        private var value: pthread_key_t
-#endif
-
-#if os(macOS) || os(iOS) || os(tvOS) || os(watchOS)
-        public typealias KeyDestructor =
-            @convention(c) (UnsafeMutableRawPointer) -> Void
-#else
-        public typealias KeyDestructor =
-            @convention(c) (UnsafeMutableRawPointer?) -> Void
-#endif
+    internal class Key {
+        private var value: ThreadOpsSystem.ThreadSpecificKey
```

Comment:
Can we name this `key` instead?

It's misleading in function calls: `getThreadSpecificValue(self.value)`, `setThreadSpecificValue(key: self.value, value: value)`